tests.sh

```diff
@@ -96,8 +94,9 @@ assert "_clean; x=0; assert 'x=1'; assert_raises 'x=2'; echo \$x" 0
 assert "_clean; x=0; assert 'export x=1'; assert_raises 'export x=2';
 echo \$x" 0
 # options do not leak
-assert_raises "set +e"
-assert_raises "shopt -o errexit"
+assert_raises "shopt -o errexit" 1
```

Comment:
Because of the removal of `set -e` above, `set +e` should be active now, so we first test, for that (`shopt -o errexit` returns 1 with `+e` and 0 with `-e`), then do a `set -e` inside an assertion and then we check it doesn't bleed through by making sure the next line `set +e` is still in effect.